website/project/decorators.py

```diff
@@ -247,6 +247,30 @@ def wrapped(*args, **kwargs):
 must_be_contributor_or_public_but_not_anonymized = _must_be_contributor_factory(include_public=True, include_view_only_anon=False)
 
 
+def must_be_contributor_even_if_public(func):
```

Comment:
I'm not sure that this is right approach. This file has a [decorator factory](https://github.com/CenterForOpenScience/osf.io/blob/develop/website/project/decorators.py#L194) intended to handle contributor / public and anonymous behavior. Three of the four possible behavior patterns are [defined above](https://github.com/CenterForOpenScience/osf.io/blob/develop/website/project/decorators.py#L244-L247), and it seems like the one we want for this issue is the yet-undefined one --  `(include_public=False, include_view_only_anon=False)`. 

However, `include_view_only_anon` shouldn't matter for this case. As a non-contributor is making it through the `@must_be_contributor` decorator, there's apparently a bug in the decorator factory.